test/unit/test_connection.py

```diff
@@ -136,12 +136,46 @@ def test_is_still_running():
 
 
 @pytest.mark.skipolddriver
-def test_partner_env_var():
-    with patch.dict(os.environ, {ENV_VAR_PARTNER: "Amanda"}):
-        with patch("snowflake.connector.network.SnowflakeRestful.fetch"):
-            with snowflake.connector.connect(
-                user="user",
-                account="account",
-                password="password123",
-            ) as conn:
-                assert conn.application == "Amanda"
+@patch("snowflake.connector.network.SnowflakeRestful._post_request")
+def test_partner_env_var(mockSnowflakeRestfulPostRequest, capsys):
+    PARTNER_NAME = "Amanda"
+
+    def mock_post_request(url, headers, json_body, **kwargs):
+        global mock_cnt
```

Comment:
What would this refer to? As far as I can tell there's no `mock_cnt` defined globally.
I'd move the definition up from line 167 to above the function and use `nonlocal` keyword here instead.

Reply:
Agreed. Should have paid better attention when copying from another test.

Reply:
I removed all the mock_cnt pieces, and switch to testing the data as a dictionary rather than just reading standard out